packages/integrations/mdx/src/rehype-images-to-component.ts

```diff
@@ -0,0 +1,163 @@
+import type { MarkdownVFile } from '@astrojs/markdown-remark';
+import type { Properties, Root } from 'hast';
+import type { MdxJsxAttribute, MdxjsEsm } from 'mdast-util-mdx';
+import type { MdxJsxFlowElementHast } from 'mdast-util-mdx-jsx';
+import { visit } from 'unist-util-visit';
+import { jsToTreeNode } from './utils.js';
+
+export const ASTRO_IMAGE_ELEMENT = 'astro-image';
+export const ASTRO_IMAGE_IMPORT = '__AstroImage__';
+export const USES_ASTRO_IMAGE_FLAG = '__usesAstroImage';
+
+function createArrayAttribute(name: string, values: (string | number)[]): MdxJsxAttribute {
+	return {
+		type: 'mdxJsxAttribute',
+		name: name,
+		value: {
+			type: 'mdxJsxAttributeValueExpression',
+			value: name,
+			data: {
+				estree: {
+					type: 'Program',
+					body: [
+						{
+							type: 'ExpressionStatement',
+							expression: {
+								type: 'ArrayExpression',
+								elements: values.map((value) => ({
+									type: 'Literal',
+									value: value,
+									raw: String(value),
+								})),
+							},
+						},
+					],
+					sourceType: 'module',
+					comments: [],
+				},
+			},
+		},
+	};
+}
+
+/**
+ * Convert the <img /> element properties (except `src`) to MDX JSX attributes.
+ *
+ * @param {Properties} props - The element properties
+ * @returns {MdxJsxAttribute[]} The MDX attributes
+ */
+function getImageComponentAttributes(props: Properties): MdxJsxAttribute[] {
+	const attrs: MdxJsxAttribute[] = [];
+
+	for (const [prop, value] of Object.entries(props)) {
+		if (prop === 'src') continue;
+
+		if (Array.isArray(value)) {
+			attrs.push(createArrayAttribute(prop, value));
+		} else if (prop === 'widths') {
+			attrs.push(createArrayAttribute(prop, String(value).split(' ')));
```

Comment:
I didn't realize when creating the initial issue for this that properties were all turned into "sanitizied" strings I sort of figured if given a hProperty by a remark plugin like `width: [100,200]` it would become `width="[100,200]"` but you've solved this how I initially implemented the remark version's ability to add custom properties to the image component, by putting a condition on widths.

A condition will also need to be added for `denisities` since this is also an array option, and
```
if (Array.isArray(value)) {
			attrs.push(createArrayAttribute(prop, value));
		}
```
Can be removed because it will never be true based on how unified is sanitizing things

But the reason that we implemented this Array.isArray method was to empower other image services that might make use of their own custom properties that want to accept arrays, instead of hard coding special cases for widths and denisities, but I can't exactly see another way to do this and the imageService if they wanted to make use of these properties could handing that property receiving a string of values and just `.split(' ')` it themselves for its users 

@Princesseuh What do you think?

Reply:
Changes are ready (not pushed yet, I'm waiting for any other changes).

Isn't `inferSize` likely to cause problems as well? Since it is supposed to be a boolean, `"false"` will be evaluated [as true in `imageService`](https://github.com/withastro/astro/blob/fa0f593890502faf5709ab881fe0e45519d2f7af/packages/astro/src/assets/internal.ts#L63), right? However I don't see how to convert the attribute's value to a boolean. The `value` property in `MdxJsxAttribute` is typed as `string | MdxJsxAttributeValueExpression | null | undefined`. Maybe we could use an empty string if the value is not `"true"`, something like `value: value === "true" ? value : ""`.

Reply:
As of right now remote images are not processed by `![]()` syntax, but if we did allow remote images inferSize would most likely be turned on by default and it would the need to be able to be turned off in a granular way with a remark/rehype plugin youre right but I don't think that we need to handle that here now

Reply:
Yeah, considering that remote images are not supported at this time, it's fine to not have the best support for inferSize.